pkg/registry/networking/ingressclass/declarative_validation_test.go

```diff
@@ -0,0 +1,281 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingressclass
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	apitesting "k8s.io/kubernetes/pkg/api/testing"
+	networking "k8s.io/kubernetes/pkg/apis/networking"
+)
+
+var apiVersions = []string{"v1", "v1beta1"}
+
+func TestDeclarativeValidateParametersName(t *testing.T) {
+	for _, apiVersion := range apiVersions {
+		t.Run(apiVersion, func(t *testing.T) {
+			ctx := genericapirequest.WithRequestInfo(
+				genericapirequest.NewDefaultContext(),
+				&genericapirequest.RequestInfo{
+					APIGroup:          "networking.k8s.io",
+					APIVersion:        apiVersion,
+					Resource:          "ingressclasses",
+					IsResourceRequest: true,
+					Verb:              "create",
+				},
+			)
+
+			testCases := map[string]struct {
+				input        networking.IngressClass
+				expectedErrs field.ErrorList
+			}{
+				"valid": {
+					input: mkValidIngressClass(),
+				},
+				"missing parameters.name": {
+					input: mkValidIngressClass(func(obj *networking.IngressClass) {
+						obj.Spec.Parameters.Name = ""
+					}),
+					expectedErrs: field.ErrorList{
+						field.Required(
+							field.NewPath("spec", "parameters", "name"),
+							"name is required",
```

Comment:
you don't really need the detail string here - the match on type (Required) is sufficient.